### PR TITLE
feat: add code generation evaluation with test-case-driven sandbox environment

### PR DESCRIPTION
**Pull request by brluobt**
## Background

As a new NeMo RL user, I found the GRPO Quick Start guide and evaluation pipeline excellent for getting started with math tasks. However, when I tried to extend my workflow to code generation — an increasingly important use case for RL-based LLM training — I realized there was no equivalent quick-start path:

- The existing `CodeEnvironment` provides sandbox execution but has no test-case validation or reward signal, making it unsuitable for evaluation or GRPO training out of the box.
- `run_eval.py` only works with `MathEnvironment`, so there is no simple command to benchmark a coding model.
- There are no sample configs or standard benchmarks integrated for code generation, which raises the barrier for developers who want to explore this direction.

My goal with this PR is to provide the same "build, run, see results" experience that math tasks already have, but for code generation. A developer should be able to:

1. Run a single eval command against LiveCodeBench and see a pass@k score.
2. Launch GRPO training with code test-case rewards using a sample config.
3. Use this as a starting point to plug in their own code benchmarks.

Relates to #858

## Summary

Add end-to-end code generation evaluation to NeMo RL. The pipeline supports both evaluation (pass@k on LiveCodeBench) and GRPO training with code test-case rewards.

Quick start examples:

```bash
# Evaluate a coding model (pass@1 on LiveCodeBench)
uv run examples/run_eval.py --config examples/configs/evals/code_eval.yaml

# Train with GRPO on code generation
uv run examples/run_grpo.py --config examples/configs/grpo_code.yaml cluster.gpus_per_node=8
```

**Validated end-to-end**: Qwen2.5-Coder-1.5B-Instruct on LiveCodeBench v5 (167 problems), pass@1 = 3.0% with public test cases.

### 1. CodeTestCaseEnvironment (core)
- `nemo_rl/environments/code_test_environment.py` — New environment that extracts code from model responses, executes it against stdin/stdout test cases via subprocess, and returns binary pass/fail rewards (1.0 if all tests pass, 0.0 otherwise). Follows the MathEnvironment pattern with accuracy and pass@k metrics in `global_post_process_and_metrics()`.
- `nemo_rl/environments/utils.py` — Register `"code_test"` in `ENV_REGISTRY`
- `nemo_rl/distributed/ray_actor_environment_registry.py` — Register actor with `PY_EXECUTABLES.SYSTEM`

### 2. Data Processor
- `nemo_rl/data/processors.py` — Add `code_data_processor` that maps code problems with test cases into `DatumSpec`, passing `test_cases` and `ground_truth` through `extra_env_info`. Registered in `PROCESSOR_REGISTRY`.

### 3. Generalize Eval Pipeline
- `examples/run_eval.py` — Replace hardcoded `MathEnvironment` with `create_env(env_name, ...)`. The `env_name` is read from data config and defaults to `"math"`, so **all existing math eval configs work unchanged**.

### 4. LiveCodeBench Dataset Integration
- `nemo_rl/data/datasets/eval_datasets/livecodebench.py` — Eval dataset loader supporting release_v1 through v6.
- `nemo_rl/data/datasets/response_datasets/livecodebench.py` — GRPO training dataset following the `RawDataset` pattern.
- Both loaders handle `datasets` 3.x (script loading) and 4.x (JSONL fallback) for container compatibility.

### 5. Configs and Tests
- `examples/configs/evals/code_eval.yaml` — Ready-to-run code evaluation config
- `examples/configs/grpo_code.yaml` — Ready-to-run GRPO code training config
- `examples/prompts/code.txt` — Prompt template for code generation
- `tests/unit/environments/test_code_test_environment.py` — Unit tests covering code extraction, test case execution, timeout handling, batch processing, and extracted answer support.

## Splitting Offer

This PR is self-contained but can be split if preferred:
- **PR A (core)**: Components 1–3 (CodeTestCaseEnvironment + code_data_processor + eval generalization)
- **PR B (integration)**: Components 4–5 (LiveCodeBench datasets + configs + tests)

Happy to split on request.

## Test Plan

- [x] Unit tests for code extraction (`extract_code`) and test case execution (`run_single_test`)
- [x] Ray-based integration tests for `CodeTestCaseEnvironment` (pass, fail, batch, extracted answer)
- [x] Dataset loading verified (LiveCodeBench v5, 167 samples; JSONL fallback for `datasets` 4.x)
- [x] Backward compatibility confirmed (existing math eval configs work unchanged with default `env_name="math"`)
- [x] End-to-end eval in Docker container: Qwen2.5-Coder-1.5B-Instruct on H20, pass@1 = 3.0% (5/167)
- [ ] GRPO training convergence validation (planned follow-up)

## Summary by CodeRabbit

* **New Features**
  * Code generation evaluation config and GRPO training config added
  * LiveCodeBench integrated for training and evaluation (dataset + prompts)
  * New code-testing environment to run and validate Python solutions
  * Data processor and registry updates to support code evaluation workflows
  * Eval workflow updated to create environments dynamically

* **Tests**
  * Added comprehensive unit tests for the code testing environment

### CODE REVIEW

**Comment by brluobt** [comment]
@coderabbitai review

Pushed a fix commit addressing all 11 review comments (schema mismatch, subprocess hardening, zip strict, type checks, copyright, docstrings, naming conventions). Please re-review.
